src/App.tsx

```diff
@@ -2,10 +2,46 @@ import React from 'react'
 import logo from './logo.svg'
 import './App.css'
 import FileImportButton from './Components/FileImportButton'
+import TextField from '@material-ui/core/TextField'
+import xml2js from 'xml2js'
+import Attempt from './types'
 
 const App = () => {
-  function handleFile() {
-    console.log('test')
+  const [attemptHistory, setAttemptHistory] = React.useState<Attempt[]>([])
+
+  function handleFile(e: React.ChangeEvent<HTMLInputElement>) {
+    if (e?.target?.files?.[0]) {
```

Comment:
You might consider:
```
if (!e?.target?.files?.[0]) {
  return null
}
... rest of function
```
As a quick escape in order to reduce the length of the if-case and to un-indent the rest of the method body.

Reply:
changed